zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupUpdateProcessor.java

```diff
@@ -0,0 +1,106 @@
+import io.camunda.zeebe.engine.state.distribution.DistributionQueue;
+import io.camunda.zeebe.engine.state.immutable.GroupState;
+import io.camunda.zeebe.protocol.impl.record.value.group.GroupRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.GroupIntent;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+
+public class GroupUpdateProcessor implements DistributedTypedRecordProcessor<GroupRecord> {
+
+  private final GroupState groupState;
+  private final KeyGenerator keyGenerator;
+  private final AuthorizationCheckBehavior authCheckBehavior;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+  private final CommandDistributionBehavior commandDistributionBehavior;
+
+  public GroupUpdateProcessor(
+      final GroupState groupState,
+      final KeyGenerator keyGenerator,
+      final AuthorizationCheckBehavior authCheckBehavior,
+      final Writers writers,
+      final CommandDistributionBehavior commandDistributionBehavior) {
+    this.groupState = groupState;
+    this.keyGenerator = keyGenerator;
+    this.authCheckBehavior = authCheckBehavior;
+    this.commandDistributionBehavior = commandDistributionBehavior;
+    stateWriter = writers.state();
+    responseWriter = writers.response();
+    rejectionWriter = writers.rejection();
+  }
+
+  @Override
+  public void processNewCommand(final TypedRecord<GroupRecord> command) {
+    final var record = command.getValue();
+    final var persistedRecord = groupState.get(record.getGroupKey());
+    if (persistedRecord.isEmpty()) {
+      final var errorMessage =
+          "Expected to update group with key '%s', but a group with this key does not exist."
+              .formatted(record.getGroupKey());
+      rejectionWriter.appendRejection(command, RejectionType.NOT_FOUND, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.NOT_FOUND, errorMessage);
+      return;
+    }
+
+    final var groupName = record.getName();
+    final var authorizationRequest =
+        new AuthorizationRequest(command, AuthorizationResourceType.GROUP, PermissionType.UPDATE)
+            .addResourceId(groupName);
+    if (!authCheckBehavior.isAuthorized(authorizationRequest)) {
+      final var errorMessage =
+          UNAUTHORIZED_ERROR_MESSAGE.formatted(
+              authorizationRequest.getPermissionType(), authorizationRequest.getResourceType());
+      rejectionWriter.appendRejection(command, RejectionType.UNAUTHORIZED, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.UNAUTHORIZED, errorMessage);
+      return;
+    }
+
+    final var groupKey = groupState.getGroupKeyByName(groupName);
```

Comment:
❓ Is this step necessary ? Is it not enough to say that if the provided group name is the same of the persisted record the processor should reject the command ?

Reply:
That's a very good point, thanks!

I'll update this processor and the`RoleUpdateProcessor` to avoid this extra state access.

Reply:
@nicpuppa @koevskinikola This was necessary. If the group name changed we must check the state that there is not already another group with this name. Checking it with the provided record is not the same thing.

Reply:
What do you mean ? We check the provided with the record in the state

Reply:
> What do you mean ? We check the provided with the record in the state

We get the Group from the state using the key provided in the command. Then we check if the name in our state is the same as the name of the provided command.

What the check should do is make sure there are no other Groups (with different keys) that have the same name. The name should be unique.

Reply:
So basically:

```java
if (groupState.getByName(name).isPresent()) {
   // reject because a group with this name already exists.
}
```

Reply:
Ok got it, make sense 👍

Reply:
@nicpuppa Will you create a fix for it?

Reply:
yes